MapboxMobileEvents/MMENSURLSessionWrapper.m

```diff
@@ -39,6 +39,7 @@ - (void)processRequest:(NSURLRequest *)request completionHandler:(void (^)(NSDat
             dataTask = nil;
         }];
         [dataTask resume];
+        [session finishTasksAndInvalidate];
```

Comment:
If you call `finishTasksAndInvalidate` here won't that finish the dataTask above that was just `resume`d?

Reply:
As I understand it, this call will wait until the task finishes and _then_ will invalidate the session object. 

From the docs:
> existing tasks continue until completion. After the last task finishes and the session makes the last delegate call related to those tasks, the session calls the urlSession(_:didBecomeInvalidWithError:) method on its delegate, then breaks references to the delegate and callback objects. After invalidation, session objects cannot be reused.

https://developer.apple.com/documentation/foundation/urlsession/1407428-finishtasksandinvalidate

Reply:
Note that it but doesn't block the caller while waiting: `This method returns immediately without waiting for tasks to finish`